### DISCUSSION TITLE
CSP evaluator says our site does not have CSP but we can see them in header

### DISCUSSION THREAD

**Discussion by BowTiedFirefox** [vercel/community#6544] [Help]
### Summary

We follow the code example here to add CSP to headers
https://nextjs.org/docs/pages/building-your-application/configuring/content-security-policy#without-nonces
We can see the CSP headers in the firefox/chrome
But when we use scanners like https://csp-evaluator.withgoogle.com/, https://observatory.mozilla.org/
They claim we do not have CSP enable
We deploy our site on Vercel.
Does any one have similar problems and what the solution is?

### Steps to Reproduce

<img width="1426" alt="Screen Shot 2024-04-17 at 12 43 29 PM" src="https://github.com/vercel/community/assets/94278106/f2cfc5a5-7102-4dbc-a4b2-e4d246d0590a">
We can clear see CSP headers

**Comment by versecafe** [comment]
@BowTiedFirefox Can you test if the CSP is functioning by trying to inject something like Google Analytics or the Vercel toolbars to see if it effectively blocking them, if so than the CSP evaluator may just be wrong.

**Reply by BowTiedFirefox** [reply]
<img width="1430" alt="Screen Shot 2024-04-17 at 4 13 14 PM" src="https://github.com/vercel/community/assets/94278106/e28704e6-da58-4013-95d6-27146ab2a71c">
It seems it is blocking Vercel live.
We are running a preview build. Is that an issue?

**Reply by versecafe** [reply]
@BowTiedFirefox Not an issue at all actually, this is exactly what should occur if your CSP is being respected, so your CSP is functioning correctly, the scanner for it is having the issue not you.

**Reply by BowTiedFirefox** [reply]
it seems it takes a few days for result to be updated
